src/main/java/com/shapesecurity/bandolier/es2017/transformations/VariableCollisionResolver.java

```diff
@@ -20,18 +20,19 @@ private VariableCollisionResolver() {
 
 	public static Tuple3<HashTable<String, Module>, VariableNameGenerator, HashTable<Module, HashTable<String, String>>> resolveCollisions(@Nonnull HashTable<String, Module> modules) {
 		HashTable<String, GlobalScope> globalScopes = modules.foldLeft((acc, pair) -> acc.put(pair.left, ScopeAnalyzer.analyze(pair.right)), HashTable.emptyUsingIdentity());
-		HashTable<String, ImmutableSet<String>> names = modules.foldLeft((acc, module) -> GlobalVariableExtractor.extractAllDeclaredVariables(module.right, globalScopes.get(module.left).fromJust(), new ScopeLookup(globalScopes.get(module.left).fromJust())).entries().map(pair -> pair.left).foldLeft((subAcc, name) -> subAcc.put(name, module.left), acc), MultiHashTable.<String, String>emptyUsingEquality()).toHashTable(ImmutableList::uniqByEquality);
+		HashTable<String, ImmutableSet<String>> globalNames = modules.foldLeft((acc, module) -> GlobalVariableExtractor.extractAllDeclaredVariables(module.right, globalScopes.get(module.left).fromJust(), new ScopeLookup(globalScopes.get(module.left).fromJust())).entries().map(pair -> pair.left).foldLeft((subAcc, name) -> subAcc.put(name, module.left), acc), MultiHashTable.<String, String>emptyUsingEquality()).toHashTable(ImmutableList::uniqByEquality);
+		HashTable<String, ImmutableSet<String>> allNames = modules.foldLeft((acc, module) -> VariableReferenceExtractor.extractAllReferencedVariables(module.right, globalScopes.get(module.left).fromJust(), new ScopeLookup(globalScopes.get(module.left).fromJust())).entries().map(pair -> pair.left).foldLeft((subAcc, name) -> subAcc.put(name, module.left), acc), MultiHashTable.<String, String>emptyUsingEquality()).toHashTable(ImmutableList::uniqByEquality);
```

Comment:
It looks like you only need the keys here. If that's so, you shouldn't build the whole hash table.

Reply:
And in fact `VariableReferenceExtractor.extractAllReferencedVariables` can probably be simplified to just give the strings.